### PR TITLE
Feat add cellRenderer (successor of #2942)

### PR DESCRIPTION
**Pull request by Liooo**
This PR is a successor of https://github.com/adazzle/react-data-grid/pull/2942, as it's not looking active anymore.
The following 3 commits are added from the original PR:

1. ffa2b23 merging latest main to the PR
2. c3068ef fix some typing 
3. 0e614dc allow overwriting `style` for cellRenderer 

The reason behind the 3rd commit is, it was necessary to accomplish animated row reordering using https://github.com/clauderic/dnd-kit like below (this is my specific usecase but it should be useful in other scenarios as well):

dnd-kit-row-order

### CODE REVIEW

**Comment by Liooo** [comment]
@amanmahajan7 @nstepien 

Sorry for bothering, please take a look as it's almost there 🙏

**Comment by Liooo** [comment]
ping @amanmahajan7 @nstepien 

plz 🙇🙇🙇

File: src/Cell.tsx [hunk 1]
```diff
@@ -88,16 +94,25 @@ function Cell<R, SR>({
...
       role="gridcell"
       aria-colindex={column.idx + 1} // aria-colindex is 1-based
       aria-colspan={colSpan}
       aria-selected={isCellSelected}
       aria-readonly={!isEditable || undefined}
+      ref={ref}
       tabIndex={tabIndex}
       className={className}
-      style={getCellStyle(column, colSpan)}
+      style={cellStyle}
```
[the remark this answers is not in the document]
Comment (Liooo) [reply src/Cell.tsx]: @amanmahajan7 applied in [35004a0](https://github.com/adazzle/react-data-grid/pull/3621/commits/35004a074f5cd337c2d35a2e3e11cb1cd3c9d37d) 🙏

**Review by amanmahajan7 — APPROVED** [review]
Thanks for the PR

**Pull request merged**